packages/carbon-web-components/src/components/pagination-nav/pagination-nav.ts

```diff
@@ -0,0 +1,252 @@
+import { carbonElement as customElement } from '../../globals/decorators/carbon-element';
+import settings from 'carbon-components/es/globals/js/settings';
+import styles from './pagination-nav.scss';
+import { classMap } from 'lit-html/directives/class-map';
+
+const { prefix } = settings;
+
+function calculateCuts(
+  page: number,
+  totalItems: number,
+  itemsDisplayedOnPage: number,
+) {
+  const split = (itemsDisplayedOnPage - 1) / 2;
+
+  let frontHiddenIndex = page - Math.ceil(split);
+  let backHiddenIndex = page + Math.floor(split) - 1;
+
+
+  if (frontHiddenIndex < 1) {
+    backHiddenIndex += Math.abs(frontHiddenIndex - 1);
+  }
+
+  if (backHiddenIndex > totalItems - 1) {
+    frontHiddenIndex -= (Math.abs(totalItems - backHiddenIndex - 1));
+  }
+
+  return {
+    front: frontHiddenIndex,
+    back: backHiddenIndex
+  };
+}
+
+/**
+ * Pagination Navigation.
+ *
+ * @element bx-pagination
+ */
+@customElement(`${prefix}-pagination-nav`)
+class BXPaginationNav extends LitElement {
+
+  /**
+   * The maxiumum number of items to show in pager nav.
+   *   - includes overflow items
+   *   - excludes prev/next buttons
+   *   - minumum value of 5
+   */
+  @property({ attribute: 'visible', reflect: true, type: Number})
+  visible = 1;
+
+  /**
+   * Whether to allow looping between first/last pages.
+   */
+  @property({ attribute: "loop", type: Boolean})
+  loop = false;
+
+  /**
+   * Active page index (0-based).
+   */
+  @property({ attribute: "page", reflect: true, type: Number })
+  page = 0;
+
+  /**
+   * Total number of pages
+   */
+  @property({ attribute: 'count', reflect: true, type: Number})
+  count = 1;
+
+  /**
+   * Array allowing us to map out the items
+   *   - Prefer showing this.count
+   *   - Maximum of this.visible
+   *   - Minumum of 1
+   */
+  private _iterator: any[] = [];
+
+  shouldUpdate(changedProperties) {
+    // Prevent setting "visible" to less than 5.
+     if (changedProperties.has('visible') && this.visible < 5) {
+      this.visible = 5;
+    }
+
+    // Prevent setting "page" outside bounds of available pages.
+    if (changedProperties.has('count') || changedProperties.has('page')) {
+      if (this.page > this.count) {
+        this.page = this.count - 1;
+      }
+      if (this.page < 0) {
+        this.page = 0;
+      }
+    }
+
+    return true;
+  }
+  /**
+   * After values are updated, but before rendering, update the iterator.
+   *
+   * @param changedProperties map of previous values of changed properties
+   */
+  update(changedProperties) {
+    if (changedProperties.has('count') || changedProperties.has('visible') || changedProperties.has('page')) {
+      const { count, visible: visibleLimit, page } = this;
+      const allItems = [...new Array(Math.max(count, 1))].map((_u, i) => i);
+      let iterator;
+
+      if (allItems.length > visibleLimit) {
+        const first = allItems.shift();
+        const last = allItems.pop();
```

Comment:
In the React version, first and last can end up being a group. Doesn't seem like that's possible here?

![image](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/191049/4fc02af1-50dd-4bd7-b462-a8e60985790b)

Reply:
I'm pretty sure this is _not_ the intended functionality. @kennylam can you weigh in on this?

Reply:
Reached out to the carbon-react team for guidance here: https://ibm-marketing-comms.slack.com/archives/C2K6RFJ1G/p1712670269654919

Reply:
https://github.com/carbon-design-system/carbon/issues/15830#issuecomment-2030066687